src/shards.rs

```diff
@@ -173,17 +173,14 @@ impl VoiceUpdate for Shard {
         match self {
             #[cfg(feature = "serenity")]
             Shard::Serenity(handle) => {
-                let map = json!({
-                    "op": 4,
-                    "d": {
-                        "channel_id": channel_id.map(ChannelId::get),
-                        "guild_id": guild_id.get(),
-                        "self_deaf": self_deaf,
-                        "self_mute": self_mute,
-                    }
-                });
-
-                handle.send(ShardRunnerMessage::Message(map.to_string().into()))?;
+                handle.send(ShardRunnerMessage::UpdateVoiceState {
+                    guild_id: guild_id.get().into(),
+                    channel_id: channel_id
+                        .map(ChannelId::get)
+                        .map(serenity::all::ChannelId::from),
```

Comment:
Why the roundtrip from `ChannelId` to `u64` and back again?

Reply:
Songbird defines its own ID newtypes to be independent of whichever gateway a user wants, and the input is in our ID format. We have a From impl from serenity and twilight types to ours, but not apparently the opposite direction (hence the transformation here, I guess).